modules/reitit-frontend/src/reitit/frontend.cljs

```diff
@@ -16,10 +17,20 @@
 
 (defn match-by-path
   "Given routing tree and current path, return match with possibly
-  coerced parameters. Return nil if no match found."
+  coerced parameters. Returns nil if no match found."
   [router path]
-  (let [uri (.parse Uri path)]
-    (if-let [match (r/match-by-path router (.getPath uri))]
+  (let [uri (.parse Uri path)
+        path (.getPath uri)]
+    (if-let [match (or (r/match-by-path router path)
```

Comment:
This will make the lookup much slower as there is an extra hash-lookup for each routing call. Would be better to move the check into router creation time and return a reified `r.c/Router` with different code for the different cases.

Reply:
Still think this would be cleaner to be implemented as a reified Proxy, but that could be done later too.